backend/src/tests/tests.test.js

```diff
@@ -0,0 +1,191 @@
+const { createTestClient } = require('apollo-server-testing');
+const { ApolloServer, gql } = require("apollo-server");
+const {beforeEach, describe, expect, it} = require("@jest/globals");
+const PostsAPI = require("../datasources/postApi");
+const UsersAPI = require("../datasources/userApi");
+const typeDefs = require('../typeDefs');
+const resolvers = require('../resolvers');
+
+let usersApi = new UsersAPI();
+let postsApi = new PostsAPI();
+
+const server = new ApolloServer({
+    typeDefs, 
+    resolvers, 
+    dataSources: () => {
+        return {
+          usersApi: usersApi,
+          postsApi: postsApi
+        }
+    }
+});
+
+const { query, mutate } = createTestClient(server);
+
+describe("query", () => {
+
+    it("posts with nested user object", async () => {
+        const GET_POSTS = gql`
```

Comment:
Since `GET_POSTS` never changes in your test, you could also declare it once outside your `it` test cases.